src/librustc_middle/ty/layout.rs

```diff
@@ -2001,7 +2001,8 @@ where
                 }
 
                 let fields = match this.ty.kind {
-                    ty::Adt(def, _) => def.variants[variant_index].fields.len(),
+                    ty::Adt(def, _) =>
```

Comment:
this was actually reachable?

Reply:
I've updated `destructure_const` so that it doesn't invoke `operand_downcast` (and my extension, this function), so I've removed this change and left a assert.